policy/executor/internal/nodes.go

```diff
@@ -522,21 +522,32 @@ func (nodeData *ReportingJobNodeData) score() (*policy.Score, error) {
 		}
 	}
 
-	if nodeData.isQuery {
-		// if this reporting job represents a reporting query, we want to add that
-		// datapoints calculation, but return the score as is
-
-		if nodeData.childScores[nodeData.queryID] == nil {
-			panic("invalid reporting job")
-		}
+	if nodeData.forwardScore {
 		var s *policy.Score
-		if v := nodeData.childScores[nodeData.queryID].score; v == nil {
+
+		if len(nodeData.childScores) == 0 {
 			s = &policy.Score{
```

Comment:
This is needed because reporting query nodes in our graph don't have children if they're data queries only.